r/R/query-engine.R

```diff
@@ -191,7 +192,7 @@ ExecPlan <- R6Class("ExecPlan",
       }
       node
     },
-    Run = function(node) {
+    Run = function(node, explain = FALSE) {
```

Comment:
IIUC this is a really bad idea: you're evaluating the whole query just to print it.

Reply:
As I understand it, Acero is a push model, not pull. So when you call start producing, it fires up, doesn't matter that you haven't pulled the first batch, it's pushing batches to the reader. I also wouldn't assume that just because the RBR object goes out of scope that that sends any signal to stop producing. Assumptions aside, we should be able to observe this if we test with real data. 

But I would think that printing the query shouldn't trigger any evaluation on the data--anything >0 data being read is probably too much.

Reply:
Creating an exec plan: None
Calling StartProducing: All of it

You can call StopProducing after you call StartProducing but that isn't great right now (we still fully consume whatever files we happen to be reading at the moment) and will hopefully get better.